src/ParquetViewer.Engine/ParquetEngine.Processor.cs

```diff
@@ -146,13 +147,58 @@ public async Task<DataTable> ReadRowsAsync(List<string> selectedFields, int offs
                     }
                 }
 
-                datarow[fieldIndex] = value ?? DBNull.Value;
+                datarow[fieldIndex] = FixDateTime(value, field) ?? DBNull.Value;
 
                 rowIndex++;
                 progress?.Report(1);
             }
         }
 
+        /// <summary>
```

Comment:
I shared [some details](https://github.com/mukunku/ParquetViewer/issues/88#issuecomment-1699450575) on your issue ticket, I believe the timestamp field is malformed which is why it's being shown as a epoch value instead of datetime.

I added the patch so we can still open such fields in the app for now.

Reply:
Good point. I am wondering which package doesn't write away the metadata correctly.

It's made with `pandas`, but that is juste a wrapper to `pyarrow` - the python implementation of `Apache Arrow`, a package from the creators of `Apache Parquet`.

So strange... or it's on the `parquet-dotnet` side.

I'll see if I need to open another bug somewhere else.

Thanks